modules/light-clients/08-wasm/types/client_state.go

```diff
@@ -121,8 +121,9 @@ func (cs ClientState) Initialize(ctx sdk.Context, cdc codec.BinaryCodec, clientS
 	}
 
 	payload := InstantiateMessage{
-		ClientState:    &cs,
-		ConsensusState: consensusState,
+		ClientState:    cs.Data,
```

Comment:
If we do this, how would the contract verify the initial latest height?

We could also use `clienttypes.MarshalClientState` in order to address this issue and pass the entire client state. I'd be in favour of this because in the previous api, we were passing the entire client state.

Reply:
Looks like the contract doesn't verify the latest height that we pass at the moment, and it actually just [overrides it with the latest height in from the underlying client state](https://github.com/misko9/centauri/blob/master/light-clients/ics10-grandpa-cw/src/client.rs#L306). I believe the latest height field in the 08-wasm wrapping type is just there as a convenience, so that when you query for client states you can easily see what the latest height the chain has for that particular 08-wasm client.

Reply:
See my [other comment](https://github.com/cosmos/ibc-go/pull/5154#discussion_r1401805306)